mantidproject/mantid
# Improve workflow in Elwin tab of Data Manipulation Interface

### Description of work
This PR improves the workflow and data handling in the `Elwin` tab of the `Data Manipulation` interface. It fixes issues #36372 and #36460 , and builds upon the changes done on the `Elwin` tab on #33319. 
In summary: 
- The `Files` tab of the `Elwin` interface has been removed in favour of the `Workspaces` tab, which now can load multiple workspaces at once as well as files. 
- Option to change between two modes of display in the workspace table has been added: Displaying one row per spectra per workspace, or displaying one row per workspace. It is possible to change dynamically between the two modes with a `Checkbox`. 

Additionally, I have detected other fixes that will benefit the interface: 
- Small cosmetic changes (Preview spectrum `combobox` and `spinbox` are slightly larger in size)
- Fix a bug where preview plot is not properly updated whenever spectrum selector is a `Combobox` instead of a `SpinBox`. 
- `Group Input` Checkbox has been renamed as `Group Output` and moved next to the `Run` button. Now when ticked, the input files used in `Elwin` algorithm are grouped as `IDA_Elwin_Input` and the output files are also grouped as `IDA_Elwin_Output`, while as before only the former were grouped. 
#### Summary of work




Fixes #36372, #36460. 


### To test:

- Build dev-docs from this branch, make sure they are no formatting issues, and follow updated test instructions. 
- On second test of test instructions, once two workspaces are added to the interface (`irs26173_...` and `irs26174_...`), test that checkbox `CollapseRows` is working as intended. When clicking on checkbox, display of workspaces changes from workspace-indexed rows to domain-indexed rows. 
- Check that when clicking `Run` with `Group Output` checkbox ticked, the resulting output workspaces are grouped. 



_Added release notes_



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
